### DISCUSSION TITLE
SMOPS  soil moisture CDF file generation

### DISCUSSION THREAD

**Discussion by Anusreegk4298** [NASA-LIS/LISF#1254] [Q&A]
Hi,

I'm trying to generate an SMOPS ASCAT observation-based CDF file for the years 2008,2009,2010 and 2011. The files are being read and LDT run is finished. The *.nc file and *domain.nc files are generated.While opening using ncview, the *domain.nc file is showing the area but while plotting the *.nc file using GrADS, the expected plot is not being generated .The same was getting generated properly for the LSM OL step and the test case.
Observation count threshold was given as "30" while the temporal resolution was set to "yearly"
Please find the associated files in the attached link.
https://drive.google.com/drive/folders/1aSIt3NXJB53bHRzELwNBgH4WkbacrIKw?usp=sharing
Kindly let me know the reason behind the above issue.

Thank you
Anusree GK

**Comment by dmocko** [comment]
Hi @Anusreegk4298 

Thanks for sharing your files.  Could you also please share your ldtlog file?

It seems that LDT is not reading the SMOPS ASCAT files.  There should be lines in the ldtlog file listing the files being read.

I am not aware of 6-hourly SMOPS ASCAT files before November 2013.  Before then, we only have daily data.

Can you please list a few filenames that you have in this directory?

`SMOPS soil moisture observation directory:  ../RTSMOPS/`

Also, it appears that this config entry may be incorrect:

`SMOPS soil moisture version:              "2"`

Please see the LDT Users' Guide.  Acceptable entries for this config are: "date-based", "1.3", "2.0", "3.0", and “NESDIS V3.0 REGEN”.

**Reply by Anusreegk4298** [reply]
Hi @dmocko 

Thanks for your reply.I tried changing the  SMOPS soil moisture version to "date-based" but still couldn't generate the desired cdf plot.

some of the file names in the directory that you have asked are as follows:
NPR_SMOPS_CMAP_D20080101.gr2  
NPR_SMOPS_CMAP_D20080102.gr2   
NPR_SMOPS_CMAP_D20080103.gr2  
NPR_SMOPS_CMAP_D20080104.gr2  

The associated files along with the ldtlog is attached in the link below:
https://drive.google.com/drive/folders/1aSIt3NXJB53bHRzELwNBgH4WkbacrIKw?usp=sharing.

Kindly suggest me the necessary corrections.

Thank you 
Anusree

**Reply by dmocko** [reply]
Hi @Anusreegk4298 

I deleted the other duplicate discussion thread.

Can you please also share your ldt.config file and the .xdf file and GrADS script that you used to generate this plot?

**Reply by Anusreegk4298** [reply]
Hi @dmocko 

Thanks for the reply.I have uploaded the ldt.config file under the name "ldt.config.ascatobs_cdf". All the associated files including the .xdf file and  GrADS script  are uploaded in the link below.

https://drive.google.com/drive/folders/1aSIt3NXJB53bHRzELwNBgH4WkbacrIKw

Kindly suggest me the necessary corrections.

Thank you
Anusree

**Comment by dmocko** [comment]
Thanks for sharing.  You need to update this one line in your ".xdf" file:

`XDEF  ngrid            44802 LINEAR             1.0    0.01`

That is because the .cdf file you are opening has 44802 ngrid (seen via "ncdump").

In your GrADS script, I just changed this one line to: `'set x 4'`

After I did so, I got the resulting figure:
![smops_cdf](https://user-images.githubusercontent.com/38500701/215844254-777b82d2-7f0c-451e-a15e-b08a32482704.png)

I am not a DA expert, so I'm not sure why setting x to 100 does not give a reasonable plot.  If you try `'set x 1 100'` - or any other ~100 interval range of ngrids - you will see multiple curves.  Don't try to plot all 44802 ngrids, though, as GrADS will plot all lines, but take a very long time.

**Reply by LIS-navari** [reply]
Hi @Anusreegk4298 @dmocko 

Everywhere we have white areas (# of obs does not meet the threshold) we do not have CDF. 

![Screen Shot 2023-01-31 at 1 17 14 PM](https://user-images.githubusercontent.com/43690895/215847970-8e7ca183-ad40-4014-826d-c11a5e886e6c.png)

**Comment by Anusreegk4298** [comment]
Thank you!! @dmocko
